performance_test/perf_test_producer.ts

```diff
@@ -27,36 +27,39 @@ export class PerfTestProducer {
     const publisher = await this.connection.declarePublisher(this.publisherParams)
     publisher.on("publish_confirm", (err, confirmedIds) => {
       if (err) {
-        console.log(err)
+        this.logger.error(err)
       }
       this.metrics.addCounter("confirmed", confirmedIds.length)
     })
 
     this.displayTimer = setInterval(() => {
       this.displayMetrics()
       this.metrics.setStart()
-    }, 1000)
+    }, 500)
 
     await this.send(publisher)
+
+    return true
   }
 
   private displayMetrics(stop: boolean = false) {
     const metrics = { ...this.metrics.getMetrics(), total: this.ctr }
-    this.logger.info(`${new Date().toISOString()} - ${inspect(metrics)}`)
+    this.logger.info(`${inspect(metrics)}`)
     if (stop && this.displayTimer) {
-      clearInterval(this.displayTimer)
+      clearInterval(this.displayTimer[Symbol.toPrimitive]())
```

Comment:
Never encountered this, how does this line work? Just wondering

Reply:
it's a way to get the id of the timer. Fixing the type definition to avoid this